static/app/components/modals/sudoModal.tsx

```diff
@@ -56,23 +81,37 @@ class SudoModal extends React.Component<Props, State> {
       return;
     }
 
-    this.setState({busy: true}, () => {
+    this.setState({busy: true, isFirstStep: true}, () => {
       retryRequest().then(() => {
         this.setState({busy: false}, closeModal);
       });
     });
   };
 
-  handleError = () => {
-    this.setState({busy: false, error: true});
+  handleError = err => {
+    let errType = '';
+    if (err.status === 403) {
+      errType = 'invalidPassword';
+    } else if (err.status === 401) {
+      errType = 'invalidSSOSession';
+    } else if (
+      err.responseText ===
+      '{"superuserReason":["Ensure this field has at least 4 characters."]}'
+    ) {
+      errType = 'invalidReason';
+    }
+    this.setState({busy: false, error: true, isFirstStep: true, errorType: errType});
   };
 
   handleU2fTap = async (data: Parameters<OnTapProps>[0]) => {
     this.setState({busy: true});
 
-    const {api} = this.props;
+    const {api, superuser} = this.props;
```

Comment:
i was initially unclear of what `superuser` here is.

Can we update this to be `isSuperuser` if it's a boolean? Maybe out of scope for this pr....

Reply:
i'll look into it more as i thought superuser was pretty unclear too.